lib/config/convict.js

```diff
@@ -0,0 +1,835 @@
+      doc: 'Set to include subdomains in HSTS',
+      format: 'Boolean',
+      env: 'CMD_HSTS_INCLUDE_SUBDOMAINS',
+      default: true,
+    },
+    preload: {
+      doc: 'whether to allow preloading of the site\'s HSTS status (e.g. into browsers)',
+      format: 'Boolean',
+      env: 'CMD_HSTS_PRELOAD',
+      default: true
+    }
+  },
+  csp: {
+    enable: {
+      doc: 'whether to enable Content Security Policy.',
+      format: 'Boolean',
+      env: 'CMD_CSP_ENABLE',
+      default: true
+    },
+    directives: {
+      doc: 'Additional CSP directives to the default ones required for CodiMD to work.',
+      default: {}
+    },
+    addDefaults: {
+      doc: 'Option to control whenever the recommended CSP directives should be add.',
+      format: 'Boolean',
+      default: true
+    },
+    addDisqus: {
+      doc: 'Option to disable the addition of Disqus rules to CSP. Disabling this option will break the Disqus integration.',
+      format: 'Boolean',
+      default: true
+    },
+    addGoogleAnalytics: {
+      doc: 'Option to disable the addition of Google Analytics rules to CSP. Disabling this option will break the Google Analytics integration.',
+      format: 'Boolean',
+      default: true
+    },
+    upgradeInsecureRequests: {
+      default: 'auto'
+    },
+    reportURI: {
+      doc: 'Allows to add a URL for CSP reports in case of violations.',
+      format: '*',
+      default: undefined
+    }
+  },
+  protocolUseSSL: {
+    doc: 'Set to use TLS protocol for resources path (only applied when domain is set).',
+    format: 'Boolean',
+    env: 'CMD_PROTOCOL_USESSL',
+    default: false
+  },
+  useCDN: {
+    doc: 'Set to use CDN resources or not',
+    format: 'Boolean',
+    env: 'CMD_USECDN',
+    default: true
+  },
+  allowAnonymous: {
+    doc: 'Set to allow anonymous usage. (Allow guests general usage of CodiMD.)',
+    format: 'Boolean',
+    env: 'CMD_ALLOW_ANONYMOUS',
+    default: true
+  },
+  allowAnonymousEdits: {
+    doc: 'If `allowAnonymous` is `true`: allow users to select `freely` permission, allowing guests to edit existing notes.',
+    format: 'Boolean',
+    env: 'CMD_ALLOW_ANONYMOUS_EDITS',
+    default: false
+  },
+  allowFreeURL: {
+    doc: 'Set to allow new note creation by accessing a nonexistent note URL. This is the behavior familiar from [Etherpad](https://github.com/ether/etherpad-lite).',
+    format: 'Boolean',
+    env: 'CMD_ALLOW_FREEURL',
+    default: false
+  },
+  forbiddenNoteIDs: {
+    doc: 'List of note IDs that can\'t be used. Main purpose is to prevent bugs due usage of paths in free-url-mode.',
+    env: 'CMD_FORBIDDEN_NOTE_IDS',
+    default: ['robots.txt', 'favicon.ico', 'api', 'build', 'css', 'docs', 'fonts', 'js', 'uploads', 'vendor', 'views']
+  },
+  defaultPermission: {
+    doc: 'Set notes default permission (only applied on signed-in users).',
+    format: function (val) {
+      if (!/(freely|editable|limited|locked|protected|private)/.test(val)) {
```

Comment:
would probably be easier to read with `['freely', ... 'private'].includes(val)` but would be a little longer. Even more readable might be `Object.keys(Permission).includes(val)` with `Permission` from here: https://github.com/codimd/server/blob/50c80c99a4246c60026517b8632a1bbb05e1fe5c/lib/config/enum.js#L9-L16

Reply:
Sounds like an idea. We removed similar validation code before.